pkg/agent/config.go

```diff
@@ -163,14 +163,27 @@ func (c *Config) validate() error {
 		zeroTmpl  = "field %q cannot be zero"
 	)
 
+	validateBaseBillingConfig := func(cfg *billing.BaseClientConfig, key string) {
+		erc.Whenf(ec, cfg.PushEverySeconds == 0, zeroTmpl, key+".pushEverySeconds")
+		erc.Whenf(ec, cfg.PushRequestTimeoutSeconds == 0, zeroTmpl, ".pushRequestTimeoutSeconds")
+		erc.Whenf(ec, cfg.MaxBatchSize == 0, zeroTmpl, ".maxBatchSize")
```

Comment:
1. IIRC it's generally better to use `fmt.Sprint[f]` than string addition, because the fmt methods use an internal buffer pool whereas plain addition doesn't
2. Looks like the second two are missing the `key+`?

Otherwise, I think this is a clean way to do it, within the existing configuration mess we have :sweat_smile:, +1

Reply:
Fixed, thx.

I've actually deliberated for quite some time how to best change this code. The conclusion I came to is: ideally, each object should have it's own `Validate()`, but this will make printing the full key path in the error message harder, so I left it for future.